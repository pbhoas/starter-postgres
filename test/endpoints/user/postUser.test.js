const request = require('supertest');
const app = require('../../../src/app');
const { getCorrectUserData, getOmittedData } = require('../../utils/user');

const goodRequest = (body = {}) => (
  request(app)
    .post('/user')
    .send(body)
);

const requiredFields = [
  'username',
  'password',
  'email',
  'firstName',
  'lastName',
];

describe('POST /user', () => {
  it('correctly posts a user', async () => {
    await goodRequest(getCorrectUserData())
      .expect(({ body }) => {
        expect(body.ok).toBe(true);
        expect(body.user).toMatchObject(expect.objectContaining({
          id: expect.any(Number),
          username: 'inserted_user',
          email: 'inserted_user@admin.com',
          firstName: 'Inserted',
          lastName: 'User',
          password: expect.stringMatching(/^\$2/),
        }));
      })
      .expect(200);
  });
  requiredFields.forEach((field) => {
    const data = getOmittedData(field);
    it(`should return error if ${field} is not present on the payload`, async () => {
      await goodRequest(data)
        .expect(({ body }) => {
          expect(body.ok).toBe(false);
          expect(body.type).toBe('ModelValidation');
          expect(body.data).toMatchSnapshot(field);
        })
        .expect(400);
    });
  });
});
