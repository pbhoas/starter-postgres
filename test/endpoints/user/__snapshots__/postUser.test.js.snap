// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /user should return error if email is not present on the payload: email 1`] = `
Object {
  "email": Array [
    Object {
      "keyword": "required",
      "message": "is a required property",
      "params": Object {
        "missingProperty": "email",
      },
    },
  ],
}
`;

exports[`POST /user should return error if firstName is not present on the payload: firstName 1`] = `
Object {
  "firstName": Array [
    Object {
      "keyword": "required",
      "message": "is a required property",
      "params": Object {
        "missingProperty": "firstName",
      },
    },
  ],
}
`;

exports[`POST /user should return error if lastName is not present on the payload: lastName 1`] = `
Object {
  "lastName": Array [
    Object {
      "keyword": "required",
      "message": "is a required property",
      "params": Object {
        "missingProperty": "lastName",
      },
    },
  ],
}
`;

exports[`POST /user should return error if password is not present on the payload: password 1`] = `
Object {
  "password": Array [
    Object {
      "keyword": "required",
      "message": "is a required property",
      "params": Object {
        "missingProperty": "password",
      },
    },
  ],
}
`;

exports[`POST /user should return error if username is not present on the payload: username 1`] = `
Object {
  "username": Array [
    Object {
      "keyword": "required",
      "message": "is a required property",
      "params": Object {
        "missingProperty": "username",
      },
    },
  ],
}
`;
